Better unit tests

enricodias opened the pull request:
Unit tests should be simple and easy to write. The new tests use only 2 data providers containing a list of valid and invalid urls, making it easy to add new urls to be tested. Every domain from the old test suite was added to the correct data provider and the phpunit's output now shows the correct number of tests and assertions. The output itself already warns about failed tests showing what was tested and what was expected, so there is no need to implement those things in the test code.

Added: phpunit.xml containing tests configs, including coverage reports. Now we can run tests using just the command `vendor/bin/phpunit`

Updated: .gitignore to ignore test reports

Fixes #3

gabriel-felipe commented:
@enricodias Thanks, this is great!  Could you just format the code accordingly with the settings on `.editorconfig`? I'm having a few indentation issues here.

enricodias commented:
@gabriel-felipe the test are formatted using PSR-2. Why are you using an arbitrary code style?

gabriel-felipe commented:
@enricodias It's showing for me that the indentation on the tests was done using tabs instead of spaces. I was using 2 spaces indentation because of personal preference, but feel free to update .editorconfig to use 4 spaces instead as well and let's define PSR-12 as the default. I'll later add a CONTRIBUTE.md file with this kind of things :)

enricodias commented:
@gabriel-felipe my bad, my editor must have replaced the spaces with tabs without I'm noticing it. I have replaced them with spaces and updated the .editorconfig file.

PSR-12 includes all PSR-2 rules. The file is in conform with PSR-12 as well.

gabriel-felipe commented:
@enricodias No problem, it happens :) Thanks for the help here! Squashed the commits and merged this one ;)

gabriel-felipe commented:
@enricodias Also, it's probably a good idea for you to sync your fork with master again since I've squashed the commits so you can avoid going out of sync on future issues :)

enricodias commented:
@gabriel-felipe ok, but I had just created another PR with code that was already done.
